spec/AuthenticationAdapters.spec.js

```diff
@@ -1517,14 +1517,16 @@ describe('apple signin auth adapter', () => {
     const fakeGetSigningKeyAsyncFunction = () => {
       return { kid: '123', rsaPublicKey: 'the_rsa_public_key' };
     };
-    spyOn(util, 'promisify').and.callFake(() => fakeGetSigningKeyAsyncFunction);
+    const spy = spyOn(util, 'promisify').and.callFake(() => fakeGetSigningKeyAsyncFunction);
 
     const result = await apple.validateAuthData(
       { id: 'the_user_id', token: 'the_token' },
       { clientId: 'secret' }
     );
     expect(result).toEqual(fakeClaim);
     expect(jwt.verify.calls.first().args[2].algorithms).toEqual(fakeDecodedToken.header.alg);
+
+    spy.and.callThrough();
```

Comment:
I still believe we should not spy on a general method like `promisify`. These test can easily break when some unrelated code is changed in Parse Server or even in a dependency like the MongoDB adapter. This could make the bugs difficult to identify and cost a lot of time in the future to investigate.

Reply:
I know, but I don't find a better solution. It's not for lack of looking :(

Reply:
I know you already invested some time into this. Let me see if I can find another solution, otherwise we merge as is.

Reply:
Fixed in https://github.com/parse-community/parse-server/pull/8761/commits/e4502a7f3e06e27566460a66a452b2c9b0eb6481